taskweaver/llm/google_genai.py

```diff
@@ -147,12 +147,14 @@ def _chat_completion(
                 raise Exception(f"Invalid role: {msg['role']}")
 
         if stream is False:
-            response = self.model.generate_content(genai_messages, stream=False)
+            response:GenerateContentResponse = self.model.generate_content(genai_messages, stream=False)
             yield format_chat_message("assistant", response.text)
-
-        response = self.model.generate_content(genai_messages, stream=True)
-        for chunk_obj in response:
-            yield format_chat_message("assistant", chunk_obj.text)
+        else:
+            response:GenerateContentResponse = self.model.generate_content(genai_messages, stream=True)
+            response.resolve()
```

Comment:
Your implementation will wait until all the response part arrive?

Reply:
Please remove `response.resolve()` or put it inside `except`. Otherwise, it won't really be streaming.

Reply:
Yes, you are correct. It seems there might be a bug or implementation issue in the actual google library. This is from commments in the google generative ai lib: 

> GENERATE_CONTENT_RESPONSE_DOC = """Instances of this class manage the response of the `generate_content` method.
> 
>     These are returned by `GenerativeModel.generate_content` and `ChatSession.send_message`.
>     This object is based on the low level `glm.GenerateContentResponse` class which just has `prompt_feedback`
>     and `candidates` attributes. This class adds several quick accessors for common use cases.
> 
>     The same object type is returned for both `stream=True/False`.
> 
>     ### Streaming
> 
>     When you pass `stream=True` to `GenerativeModel.generate_content` or `ChatSession.send_message`,
>     iterate over this object to receive chunks of the response:
> 
>     ```
>     response = model.generate_content(..., stream=True):
>     for chunk in response:
>       print(chunk.text)
>     ```
> 
>     `GenerateContentResponse.prompt_feedback` is available immediately but
>     `GenerateContentResponse.candidates`, and all the attributes derived from them (`.text`, `.parts`),
>     are only available after the iteration is complete.
>     """